common/djangoapps/lang_pref/middleware.py

```diff
@@ -26,3 +28,14 @@ def process_request(self, request):
             # Set it to the LANGUAGE_SESSION_KEY (Django-specific session setting governing language pref)
             if user_pref:
                 request.session[LANGUAGE_SESSION_KEY] = user_pref
+            else:
+                # Set browser language if it is supported by system.
+                preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
+                lang_headers = [seq[0] for seq in parse_accept_lang_header(preferred_language)]
+                if lang_headers:
+                    # lang_header is list of tuple
+                    # language with highest priority in browser will be at zero index.
+                    browser_lang = lang_headers[0]
+                    if browser_lang in [seq[0] for seq in released_languages()]:
+                        set_user_preference(request.user, LANGUAGE_KEY, browser_lang)
```

Comment:
Just to be clear, we should obey the browser's chosen languages in the absence of an explicit preference. I just don't think that we should set the user's preference, as it is not their chosen preference but is just the best available option.